src/promptflow/promptflow/_sdk/schemas/_flow.py

```diff
@@ -54,10 +56,28 @@ class FlowSchema(BaseFlowSchema):
     node_variants = fields.Dict(keys=fields.Str(), values=fields.Dict())
 
 
+class PythonEagerFlowEntry(fields.Str):
+    """Entry point for eager flow. For example: pkg.module:func"""
+
+    default_error_messages = {
+        "invalid_entry": "Provided entry {entry} has incorrect format. "
+        "Python eager flow only support pkg.module:func format.",
+    }
+
+    def _validate(self, value):
+        super()._validate(value)
+        if not re.match(r"^[a-zA-Z0-9_.]+:[a-zA-Z0-9_]+$", value):
+            raise self.make_error("invalid_entry", entry=value)
+
+
 class EagerFlowSchema(BaseFlowSchema):
     """Schema for eager flow."""
 
-    # path to flow entry file.
-    path = LocalPathField(required=True)
-    # entry function
+    # entry point, for example: pkg.module:func
     entry = fields.Str(required=True)
+
+    def _deserialize(self, data, **kwargs):
+        data = super()._deserialize(data, **kwargs)
+        if data.get("language", FlowLanguage.Python) == FlowLanguage.Python:
+            PythonEagerFlowEntry().deserialize(data.get("entry"))
```

Comment:
looks a little wired to use deserialize here

Reply:
We only need to validate 1 field, so maybe directly add validation for all languages in `EagerFlowSchema._validate`

Reply:
Changed to `validates_schema` as offline synced.